### DISCUSSION TITLE
Refreshing Data without reloading the whole page

### DISCUSSION THREAD

**Discussion by Charismara** [vercel/next.js#51149] [Help] [answered]
### Summary

Hey Guys,

i'm using prisma to load some data from by database and pass them though my server component (see [here](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching#async-and-await-in-server-components)) to a client component which basically generate a table where users can edit the information.

How can i refresh the data after a user has changed it in the table?

**Comment by icyJoseph** [comment, marked as answer]
For now, I think you can use `server actions`, to revalidate a path? 

Something like:

```js
 async function work() {
    "use server";
 
    revalidatePath("/your-table-path");
  }
```

You can also revalidate by tag, but since you are using prisma, I guess the client is not fetch based or you don't have access to fetch to put a tag on the client.

This, AFAIK, should keep the state of other surrounding components.

https://github.com/vercel/next.js/assets/21013447/d978bacc-256d-4940-816b-87843f94f18d

Code from the video:

Client component:

```tsx
"use client";
import { useState, useTransition } from "react";

export const ClientComponent = ({ work }: { work: VoidFunction }) => {
  const [isPending, startTransition] = useTransition();
  const [counter, setCounter] = useState(0);
  return (
    <div>
      <p>Click below to</p>
      <button
        onClick={() => {
          startTransition(() => work());
        }}
      >
        {isPending ? "Executing..." : "Execute"}
      </button>
      <div>
        <button onClick={() => setCounter((x) => x + 1)}>{counter} + 1</button>
      </div>
    </div>
  );
};

```

And Server Component as Page:

```tsx
import { revalidatePath } from "next/cache";
import { ClientComponent } from "./ClientComponent";
import Link from "next/link";

let c = 0;
const getCount = () => c++;

export default async function Page() {
  const round = getCount();
  async function work() {
    "use server";
    await new Promise((n) => setTimeout(n, 1_000));
    revalidatePath("/about");
  }
  return (
    <div>
      About {round}
      <div>
        <Link href="/">/home</Link>
      </div>
      <ClientComponent work={work} />
    </div>
  );
}

```

**Reply by codinginflow** [reply]
Can you explain what the purpose of `startTransition` is here? I know what the hook is for and I know that the server action docs state that you should use it. But they don't explain why.

**Reply by icyJoseph** [reply]
Nope 😅 since this is alpha stuff I rather not. 

I had expected it to take over a potentially loading boundary, but it doesn't, and the loading boundary is not triggered anymore after 13.4.1

I had also expected that maybe, if the server actions was running, you wouldn't be able to navigate, or trigger other updates, but you can. Perhaps behind the curtain, server actions are wrapped by a transition anyway?

And yeah, you get a quick `isPending` state.

**Reply by codinginflow** [reply]
> Nope 😅 since this is alpha stuff I rather not.
> 
> I had expected it to take over a potentially loading boundary, but it doesn't, and the loading boundary is not triggered anymore after 13.4.1
> 
> I had also expected that maybe, if the server actions was running, you wouldn't be able to navigate, or trigger other updates, but you can. Perhaps behind the curtain, server actions are wrapped by a transition anyway?

Yea, I made the same observations. Although navigation **is** actually blocked while the server action is running, but that seems to be a bug and it happens with and without the useTransition.
